evm/logic/context.py

```diff
@@ -193,3 +194,26 @@ def returndatacopy(computation):
 
 def sighash(computation):
     computation.stack.push(computation.msg.sig_hash)
+
+
+def paygas(computation):
+    gasprice = computation.stack.pop(type_hint=constants.UINT256)
+
+    # Only valid if (1) triggered in a top level call and
+    # (2) not been set already during this transaction execution
+    if computation.msg.depth == 0 and computation.msg.paygas_gasprice == 0:
```

Comment:
`computation.msg.paygas_gasprice == 0` seems problematic because it disallows a gas price of zero which at least historically, has been allowed.  Maybe use `None` as the *not-set* value?